string node_id
string id
string group
